### ISSUE TITLE
Beats developer guide needs to describe how to install mage

### ISSUE THREAD

**Issue by dedemorton** [elastic/beats#14409]
Contributors' guide needs to be updated with instructions (or pointers to instructions) about installing mage.

We currently have mage as a dependency, but don't describe how to install it.

**Comment by dedemorton** [comment]
Also need to tell users to run `mage GenerateCustomBeat` from the beats directory

**Comment by philippkahr** [comment]
Hi @dedemorton 
I experienced the same issue when I started contributing to beats. I just started with Go and thus GOPATH and so on was not very clear. Maybe we should not only point to a direction where someone has a guide but instead, generate a little shell script that prepares your machine? Maybe also point out how to use vagrant to have a fully recreate able testing and developing environment. https://github.com/elastic/beats/issues/14148

I already started an issue about the contributing guidelines and found a meta issue. I think those should be linked.
https://github.com/elastic/beats/issues/14260
https://github.com/elastic/beats/issues/11102
